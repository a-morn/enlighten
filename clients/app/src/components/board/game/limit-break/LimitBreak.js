import React, { memo } from "react";
import "./LimitBreak.scss";

const easeInOutQuad = t => (t < 0.5 ? 2 * t * t : -1 + (4 - 2 * t) * t);

const LimitBreak = memo(({ level, status, max, charge, className }) => (
	<div className={`bg-gray-300 rounded ${className}`}>
		<div
			className={`limit-break limit-break--${status}`}
			style={{
				width: `${(100 * level) / max}%`,
				animation: `${status}Animation ${(1 -
					easeInOutQuad(Math.min(1, charge / 100))) *
					2}s infinite`
			}}
		></div>
	</div>
));

export default LimitBreak;
