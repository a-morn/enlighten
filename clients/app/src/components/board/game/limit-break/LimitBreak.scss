.limit-break {
  height: 18px;
  max-width: 100%;
  border-radius: 4px;
  &--charging {
    background-image: linear-gradient(
      to right,
      cornflowerblue,
      blueviolet,
      lightblue,
      darkblue
    );
    transition: 0.4s linear;
    transition-property: width;
  }
  &--charged {
    background-image: linear-gradient(
      to right,
      green,
      turquoise,
      blue,
      lightblue,
      purple,
      red
    );
    transition: 0.4s linear;
    transition-property: width, background-color;
  }
  &--decharge {
    background-image: linear-gradient(to right, lightgrey, grey, darkgrey);
    transition: 0.4s linear;
    transition-property: width, background-color;
  }
}

@keyframes chargingAnimation {
  0% {
    background-image: linear-gradient(
      to right,
      cornflowerblue,
      blueviolet,
      lightblue,
      darkblue
    );
  }
  25% {
    background-image: linear-gradient(
      to right,
      darkblue,
      cornflowerblue,
      blueviolet,
      lightblue
    );
  }
  50% {
    background-image: linear-gradient(
      to right,
      lightblue,
      darkblue,
      cornflowerblue,
      blueviolet
    );
  }
  75% {
    background-image: linear-gradient(
      to right,
      blueviolet,
      lightblue,
      darkblue,
      cornflowerblue
    );
  }
}

@keyframes chargedAnimation {
  0% {
    background-image: linear-gradient(
      to right,
      green,
      turquoise,
      blue,
      lightblue,
      purple,
      red
    );
  }
  20% {
    background-image: linear-gradient(
      to right,
      turquoise,
      blue,
      lightblue,
      purple,
      red,
      green
    );
  }
  40% {
    background-image: linear-gradient(
      to right,
      blue,
      lightblue,
      purple,
      red,
      green,
      turquoise
    );
  }
  60% {
    background-image: linear-gradient(
      to right,
      lightblue,
      purple,
      red,
      green,
      turquoise,
      blue
    );
  }
  80% {
    background-image: linear-gradient(
      to right,
      purple,
      red,
      green,
      turquoise,
      blue,
      lightblue
    );
  }
  100% {
    background-image: linear-gradient(
      to right,
      red,
      green,
      turquoise,
      blue,
      lightblue,
      purple
    );
  }
}
