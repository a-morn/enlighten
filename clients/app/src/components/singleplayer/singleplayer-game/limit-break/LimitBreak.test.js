import React from 'react'
import { create } from 'react-test-renderer'
import LimitBreak from './LimitBreak'

describe('LimitBreak component', () => {
  it('matches the snapshot', () => {
    const limitBreak = create(<LimitBreak level={50} max={100} />)
    expect(limitBreak.toJSON()).toMatchSnapshot()
  })
})
