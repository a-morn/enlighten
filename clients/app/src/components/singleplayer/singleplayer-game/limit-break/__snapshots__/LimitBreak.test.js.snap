// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LimitBreak component matches the snapshot 1`] = `
<div
  className="limit-break"
  style={
    Object {
      "width": "50%",
    }
  }
/>
`;
